<a href="https://colab.research.google.com/github/sonsuin/DataAnalytical_PersonalProject/blob/main/%ED%8C%8C%EC%9D%B4%EC%8D%AC%EC%8B%A4%EC%8A%B502_Lakers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Data 02: Lakers data
로스앤젤레스 레이커스(LA Lakers)는 미국 캘리포니아주 로스앤젤레스를 연고로 하는 NBA 서부 콘퍼런스 퍼시픽 디비전 소속 프로농구 팀으로, NBA 우승 17회를 기록한 명문팀입니다.  
이 데이터는 2008-2009 시즌의 LA 레이커스 경기에 대한 기록입니다.

Data from: http://www.basketballgeek.com/data/  

##1.데이터 둘러보기

In [1]:
#한글 폰트 설정하기
!sudo apt-get install -y fonts-nanum
!sudo fc-cache -fv
!rm ~/.cache/matplotlib -rf
#런타임 다시 시작

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libnvidia-common-460
Use 'sudo apt autoremove' to remove it.
The following NEW packages will be installed:
  fonts-nanum
0 upgraded, 1 newly installed, 0 to remove and 21 not upgraded.
Need to get 9,604 kB of archives.
After this operation, 29.5 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 fonts-nanum all 20170925-1 [9,604 kB]
Fetched 9,604 kB in 1s (11.6 MB/s)
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype

In [1]:
#기본 패키지 불러오기
import math
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

plt.style.use('seaborn')
sns.set(font_scale=1)
plt.rc('font', family='NanumBarunGothic')
plt.rcParams['font.family'] = 'NanumGothic'

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [2]:
%pwd

'/content'

In [3]:
#구글 드라이브 마운트하기
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
#데이터 불러오기
#드라이브에 올려준 csv 데이터 파일 불러오기
lakers = pd.read_csv('/content/drive/MyDrive/data/Lakers.csv')

In [7]:
#불러온 데이터 일단 보기
lakers.head()

date opponent game_type   time  period      etype team  \
0  20081028      POR      home  12:00       1  jump ball  OFF   
1  20081028      POR      home  11:39       1       shot  LAL   
2  20081028      POR      home  11:37       1    rebound  LAL   
3  20081028      POR      home  11:25       1       shot  LAL   
4  20081028      POR      home  11:23       1    rebound  LAL   

                player  result  points   type     x     y  
0                  NaN     NaN       0    NaN   NaN   NaN  
1            Pau Gasol  missed       0   hook  23.0  13.0  
2  Vladimir Radmanovic     NaN       0    off   NaN   NaN  
3         Derek Fisher  missed       0  layup  25.0   6.0  
4            Pau Gasol     NaN       0    off   NaN   NaN

<컬럼 설명>
* Date: 경기 일자  
* Opponent: 대전 팀  
* Game type: 홈경기 vs. 원정경기  
* Time: 분 : 초  
* Period: 쿼터(한 쿼터당 12분 씩, 동점일 경우 5 쿼터 진행)
* Etype: 유형(ejection / foul / free throw / jump ball / rebound / shot / sub / timeout / turnover / violation
퇴장 / 파울 / 자유투/ 점프볼/ 리바운드/ 슛/ 패스 / 타임아웃/ 턴오버/ 반칙)
* Team: 팀 구분(LAL: LA Lakers, 상대팀)  
* Player: 선수명  
* Result: 결과  
* Points: 점수  
* Type: 세부행동  
* X, Y: 상대편 팀 골대 뒤에서 바라본 X, Y 좌표. 골대의 위치는 (25, 5.25)이다.

In [8]:
#데이터 기본 정보 확인하기
lakers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34624 entries, 0 to 34623
Data columns (total 13 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   date       34624 non-null  int64  
 1   opponent   34624 non-null  object 
 2   game_type  34624 non-null  object 
 3   time       34624 non-null  object 
 4   period     34624 non-null  int64  
 5   etype      34624 non-null  object 
 6   team       34624 non-null  object 
 7   player     29231 non-null  object 
 8   result     16888 non-null  object 
 9   points     34624 non-null  int64  
 10  type       24242 non-null  object 
 11  x          13067 non-null  float64
 12  y          13067 non-null  float64
dtypes: float64(2), int64(3), object(8)
memory usage: 3.4+ MB


* 범주형 변수: *빈도frequency 계산 가능*
               opponent, game_type, period, etype, team, player, resuts, type
* 연속형 변수: *평균, 표준편차, min/max 등 수치 계산 가능*
               시간 변수(date, time), result, x, y

In [9]:
#연속형(수치) 데이터의 통계치 확인하기
lakers.describe()

date        period        points             x             y
count  3.462400e+04  34624.000000  34624.000000  13067.000000  13067.000000
mean   2.008694e+07      2.535524      0.462714     25.317670     13.430550
std    4.372405e+03      1.124725      0.849231     11.185946      8.665827
min    2.008103e+07      1.000000      0.000000      0.000000      3.000000
25%    2.008121e+07      2.000000      0.000000     20.000000      6.000000
50%    2.009012e+07      3.000000      0.000000     25.000000     10.000000
75%    2.009031e+07      4.000000      1.000000     31.000000     20.000000
max    2.009041e+07      5.000000      3.000000     51.000000     90.000000

In [11]:
#범주형 데이터의 통계치 확인하기
lakers.describe(include=np.object)

<ipython-input-11-3442eff4d532>:2: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  lakers.describe(include=np.object)


opponent game_type   time  etype   team       player result   type
count     34624     34624  34624  34624  34624        29231  16888  24242
unique       29         2    717     10     31          370      2     73
top         DEN      home   0:00   shot    LAL  Kobe Bryant   made    def
freq       1855     17471    357  13108  17235         2945   8916   4768

In [12]:
#변수형 컬럼 안에는 어떤 내용이 들어있을까?
print('Oppenent: ', lakers['opponent'].unique())
print('Game type: ', lakers['game_type'].unique())
print('Period: ', lakers['period'].unique())
print('Etype: ', lakers['etype'].unique())
print('Team: ', lakers['team'].unique())
print('Result: ', lakers['result'].unique())
print('Points: ', lakers['points'].unique())
print('Type: ', lakers['type'].unique())

Oppenent:  ['POR' 'LAC' 'DEN' 'HOU' 'DAL' 'DET' 'CHI' 'PHX' 'SAC' 'NJN' 'TOR' 'IND'
 'PHI' 'WAS' 'MIL' 'MIN' 'NYK' 'MIA' 'ORL' 'MEM' 'NOH' 'BOS' 'UTA' 'GSW'
 'SAS' 'CLE' 'OKC' 'ATL' 'CHA']
Game type:  ['home' 'away']
Period:  [1 2 3 4 5]
Etype:  ['jump ball' 'shot' 'rebound' 'foul' 'free throw' 'turnover' 'timeout'
 'sub' 'violation' 'ejection']
Team:  ['OFF' 'LAL' 'POR' 'LAC' 'DEN' 'HOU' 'DAL' 'DET' 'CHI' 'PHX' 'SAC' 'NJN'
 'TOR' 'IND' 'PHI' 'WAS' 'MIL' 'MIN' 'NYK' 'MIA' 'ORL' 'MEM' 'NOH' 'BOS'
 'UTA' 'GSW' 'SAS' 'CLE' 'OKC' 'ATL' 'CHA']
Result:  [nan 'missed' 'made']
Points:  [0 2 1 3]
Type:  [nan 'hook' 'off' 'layup' 'shooting' 'personal' 'jump' 'pullup jump' 'def'
 'driving layup' 'driving finger roll layup' 'regular' 'offensive' '3pt'
 'turnaround jump' 'putback layup' 'slam dunk' 'tip' 'dunk'
 'defensive goaltending' 'hook bank' 'running layup' 'official'
 'driving slam dunk' 'short' 'driving reverse layup' 'kicked ball'
 'putback dunk' 'technical' 'alley oop dunk' 'turnaround fa

In [13]:
lakers.shape

(34624, 13)

In [18]:
lakers.isnull().sum().sum()

76625

In [21]:
#데이터 내용 간단히 정리해서 출력하기
print("총 데이터 개수: ", lakers.shape[0]*lakers.shape[1])
print("총 결측치 수: {} = 전체 데이터의 {:.2f}% ".format(lakers.isnull().sum().sum(), (lakers.isnull().sum().sum() * 100)/(lakers.shape[0]*lakers.shape[1])))
print("LA 레이커스와 경기한 팀 수: ", lakers['opponent'].nunique()-1)
print("경기에 등장하는 행동 수: ", lakers['etype'].nunique())
print("경기에 등장하는 세부행동 수: ", lakers['type'].nunique())

총 데이터 개수:  450112
총 결측치 수: 76625 = 전체 데이터의 17.02% 
LA 레이커스와 경기한 팀 수:  28
경기에 등장하는 행동 수:  10
경기에 등장하는 세부행동 수:  73


##2.데이터에서 질문 만들기

* LA레이커스의 홈 경기 vs. 원정경기 비율은?
* 경기에서 선수들이 가장 많이 하는 행동유형(etype)은?
* 이번 시즌에서 LA레이커스의 경기 결과는?
* LA레이커스 선수들은 코트의 어느 위치에서 어떤 동작을 했는가?

##3.데이터 정비하기

###3-1. 시간 타입의 컬럼 정비하기

In [23]:
#date 컬럼은 int64, time 컬럼은 object 타입이므로 date 컬럼의 타입을 변환한 후 두 컬럼을 합친다
lakers['date'] = lakers['date'].apply(str)
lakers['datetime'] = lakers['date'] + ' ' + lakers['time']
lakers.head()

date opponent game_type   time  period      etype team  \
0  20081028      POR      home  12:00       1  jump ball  OFF   
1  20081028      POR      home  11:39       1       shot  LAL   
2  20081028      POR      home  11:37       1    rebound  LAL   
3  20081028      POR      home  11:25       1       shot  LAL   
4  20081028      POR      home  11:23       1    rebound  LAL   

                player  result  points   type     x     y        datetime  
0                  NaN     NaN       0    NaN   NaN   NaN  20081028 12:00  
1            Pau Gasol  missed       0   hook  23.0  13.0  20081028 11:39  
2  Vladimir Radmanovic     NaN       0    off   NaN   NaN  20081028 11:37  
3         Derek Fisher  missed       0  layup  25.0   6.0  20081028 11:25  
4            Pau Gasol     NaN       0    off   NaN   NaN  20081028 11:23

In [24]:
#date 컬럼과 새로 만든 datetime 컬럼을 날짜 타입으로 바꿔준다
#날짜 타입은 연, 월, 일, 시, 분, 초 등의 정보를 보기 편하다(시계열 분석시 편리!)
lakers['datetime'] = pd.to_datetime(lakers['datetime'], format='%Y%m%d %H:%M:%S')
lakers['date'] = pd.to_datetime(lakers['date'], format='%Y%m%d')
lakers.head()

date opponent game_type   time  period      etype team  \
0 2008-10-28      POR      home  12:00       1  jump ball  OFF   
1 2008-10-28      POR      home  11:39       1       shot  LAL   
2 2008-10-28      POR      home  11:37       1    rebound  LAL   
3 2008-10-28      POR      home  11:25       1       shot  LAL   
4 2008-10-28      POR      home  11:23       1    rebound  LAL   

                player  result  points   type     x     y            datetime  
0                  NaN     NaN       0    NaN   NaN   NaN 2008-10-28 12:00:00  
1            Pau Gasol  missed       0   hook  23.0  13.0 2008-10-28 11:39:00  
2  Vladimir Radmanovic     NaN       0    off   NaN   NaN 2008-10-28 11:37:00  
3         Derek Fisher  missed       0  layup  25.0   6.0 2008-10-28 11:25:00  
4            Pau Gasol     NaN       0    off   NaN   NaN 2008-10-28 11:23:00

###3-2. 결측치 처리하기

In [28]:
#NULL이 존재하는 컬럼 확인하기 
lakers.isnull().any(axis=0)
#player, result, type, x, y 컬럼에 NaN 존재한다

date         False
opponent     False
game_type    False
time         False
period       False
etype        False
team         False
player        True
result        True
points       False
type          True
x             True
y             True
datetime     False
dtype: bool

In [31]:
lakers.index[lakers['x']=='0']

Int64Index([], dtype='int64')

In [32]:
#컬럼의 특성마다 어떻게 null을 처리해줄 것인가?
#player, result, type은 object 값이 들어가 있으므로 굳이 NaN에 값을 채울 필요 없다
#x, y 컬럼의 NaN: 그냥 둬도되지만 0을 채워 넣어보자
#0을 채워 넣기 전에, 실제 데이터에 0이 존재하는지 확인!
print(lakers.index[lakers['x'] == '0'].tolist())
print(lakers.index[lakers['y'] == '0'].tolist())
#존재하지 않는다!

[]
[]


In [33]:
#x, y 포지션의 NaN에 0 채워넣기
lakers['x'].fillna(0, inplace=True)
lakers['y'].fillna(0, inplace=True)
lakers.head()

date opponent game_type   time  period      etype team  \
0 2008-10-28      POR      home  12:00       1  jump ball  OFF   
1 2008-10-28      POR      home  11:39       1       shot  LAL   
2 2008-10-28      POR      home  11:37       1    rebound  LAL   
3 2008-10-28      POR      home  11:25       1       shot  LAL   
4 2008-10-28      POR      home  11:23       1    rebound  LAL   

                player  result  points   type     x     y            datetime  
0                  NaN     NaN       0    NaN   0.0   0.0 2008-10-28 12:00:00  
1            Pau Gasol  missed       0   hook  23.0  13.0 2008-10-28 11:39:00  
2  Vladimir Radmanovic     NaN       0    off   0.0   0.0 2008-10-28 11:37:00  
3         Derek Fisher  missed       0  layup  25.0   6.0 2008-10-28 11:25:00  
4            Pau Gasol     NaN       0    off   0.0   0.0 2008-10-28 11:23:00

##4.EDA & Visualizaion

###4-1. 홈 경기 대 원정 경기의 비율

In [40]:
#날짜별로 홈 경기인지 원정경기인지 보려면?
lakers[['date','game_type']].groupby(['date','game_type']).sum()

Empty DataFrame
Columns: []
Index: [(2008-10-28 00:00:00, home), (2008-10-29 00:00:00, away), (2008-11-01 00:00:00, away), (2008-11-05 00:00:00, home), (2008-11-09 00:00:00, home), (2008-11-11 00:00:00, away), (2008-11-14 00:00:00, home), (2008-11-18 00:00:00, home), (2008-11-20 00:00:00, away), (2008-11-21 00:00:00, home), (2008-11-23 00:00:00, home), (2008-11-25 00:00:00, home), (2008-11-28 00:00:00, home), (2008-11-30 00:00:00, home), (2008-12-02 00:00:00, away), (2008-12-03 00:00:00, away), (2008-12-05 00:00:00, away), (2008-12-07 00:00:00, home), (2008-12-09 00:00:00, away), (2008-12-10 00:00:00, home), (2008-12-12 00:00:00, home), (2008-12-14 00:00:00, home), (2008-12-16 00:00:00, home), (2008-12-19 00:00:00, away), (2008-12-20 00:00:00, away), (2008-12-22 00:00:00, away), (2008-12-23 00:00:00, away), (2008-12-25 00:00:00, home), (2009-01-02 00:00:00, home), (2009-01-04 00:00:00, home), (2009-01-06 00:00:00, home), (2009-01-07 00:00:00, away), (2009-01-09 00:00:00, home), (2009-01-11 00:00:00, home), (2009-01-13 00:00:00, away), (2009-01-14 00:00:00, away), (2009-01-16 00:00:00, home), (2009-01-19 00:00:00, home), (2009-01-21 00:00:00, away), (2009-01-22 00:00:00, home), (2009-01-25 00:00:00, home), (2009-01-30 00:00:00, away), (2009-01-31 00:00:00, away), (2009-02-02 00:00:00, away), (2009-02-04 00:00:00, away), (2009-02-05 00:00:00, away), (2009-02-08 00:00:00, away), (2009-02-10 00:00:00, home), (2009-02-17 00:00:00, home), (2009-02-18 00:00:00, away), (2009-02-20 00:00:00, home), (2009-02-22 00:00:00, away), (2009-02-24 00:00:00, away), (2009-02-26 00:00:00, home), (2009-02-27 00:00:00, away), (2009-03-01 00:00:00, away), (2009-03-03 00:00:00, home), (2009-03-06 00:00:00, home), (2009-03-09 00:00:00, away), (2009-03-11 00:00:00, away), (2009-03-12 00:00:00, away), (2009-03-15 00:00:00, home), (2009-03-17 00:00:00, home), (2009-03-19 00:00:00, home), (2009-03-21 00:00:00, away), (2009-03-24 00:00:00, away), (2009-03-26 00:00:00, away), (2009-03-27 00:00:00, away), (2009-03-29 00:00:00, away), (2009-03-31 00:00:00, away), (2009-04-01 00:00:00, away), (2009-04-03 00:00:00, home), (2009-04-05 00:00:00, home), (2009-04-07 00:00:00, away), (2009-04-09 00:00:00, home), (2009-04-10 00:00:00, away), (2009-04-12 00:00:00, home), (2009-04-14 00:00:00, home)]

[78 rows x 0 columns]

In [41]:
#reset_index 함수로 정리해주기
lakers[['date','game_type']].groupby(['date','game_type']).sum().reset_index()

date game_type
0  2008-10-28      home
1  2008-10-29      away
2  2008-11-01      away
3  2008-11-05      home
4  2008-11-09      home
..        ...       ...
73 2009-04-07      away
74 2009-04-09      home
75 2009-04-10      away
76 2009-04-12      home
77 2009-04-14      home

[78 rows x 2 columns]

In [42]:
#홈 경기 대 원정 경기의 비율: 
#날짜별로 home/away 데이터가 중복되므로 처음 하나만 남기고 날려버려서 해결하는 방법도 있다
df1 = lakers.drop_duplicates(['date'], keep='first')
df1

date opponent game_type   time  period      etype team player  \
0     2008-10-28      POR      home  12:00       1  jump ball  OFF    NaN   
416   2008-10-29      LAC      away  12:00       1  jump ball  OFF    NaN   
871   2008-11-01      DEN      away  12:00       1  jump ball  OFF    NaN   
1364  2008-11-05      LAC      home  12:00       1  jump ball  OFF    NaN   
1867  2008-11-09      HOU      home  12:00       1  jump ball  OFF    NaN   
...          ...      ...       ...    ...     ...        ...  ...    ...   
32345 2009-04-07      SAC      away  12:00       1  jump ball  OFF    NaN   
32782 2009-04-09      DEN      home  12:00       1  jump ball  OFF    NaN   
33280 2009-04-10      POR      away  12:00       1  jump ball  OFF    NaN   
33746 2009-04-12      MEM      home  12:00       1  jump ball  OFF    NaN   
34153 2009-04-14      UTA      home  12:00       1  jump ball  OFF    NaN   

      result  points type    x    y            datetime  
0        NaN       0  NaN  0.0  0.0 2008-10-28 12:00:00  
416      NaN       0  NaN  0.0  0.0 2008-10-29 12:00:00  
871      NaN       0  NaN  0.0  0.0 2008-11-01 12:00:00  
1364     NaN       0  NaN  0.0  0.0 2008-11-05 12:00:00  
1867     NaN       0  NaN  0.0  0.0 2008-11-09 12:00:00  
...      ...     ...  ...  ...  ...                 ...  
32345    NaN       0  NaN  0.0  0.0 2009-04-07 12:00:00  
32782    NaN       0  NaN  0.0  0.0 2009-04-09 12:00:00  
33280    NaN       0  NaN  0.0  0.0 2009-04-10 12:00:00  
33746    NaN       0  NaN  0.0  0.0 2009-04-12 12:00:00  
34153    NaN       0  NaN  0.0  0.0 2009-04-14 12:00:00  

[78 rows x 14 columns]

In [ ]:
#홈 경기 대 원정 경기의 비율: 그래프로 나타내기
f, ax = plt.subplots(1, 2, figsize=(18, 8))

#pie plot
df1['game_type']._____________().plot.___(explode=[0, 0.1], autopct='%1.1f%%', ax=ax[_], shadow=True)
ax[_]._________('Pie plot: 홈 경기 vs. 원정 경기', size = 15)
ax[_]._________('') #y 라벨 없애기

#count plot
sns.________('game_type', data=___, ax=ax[_])
ax[_]._________('Count plot: 홈 경기 vs. 원정 경기', size = 15)

plt.show()

###4-2.경기에서 발생한 행동 유형 확인하기

In [ ]:
#몇개의 행동 유형이 있는가?
print('Etype: ', lakers['etype'].________())

In [ ]:
#모든 경기를 포함해서 각각의 행동유형이 몇번 발생했는지 세기
lakers['etype'].______________()

In [ ]:
#행동유형 발생 횟수 그래프로 나타내기
f, ax = plt.________(1, 1, figsize=(14, 6))

sns._________(x=lakers['etype'], ax=ax)
ax.__________('행동 유형 발생 횟수', size=15)
ax.set_ylabel('')

plt.show()

In [ ]:
#특정 쿼터에서 많이 발생한 행동이 있을까?
#groupby를 사용해서 각 쿼터별로 각각의 행동이 몇 회 발생했는지 세 보기
df2 = lakers.________(['_____','______']).____().reset_index()._____(columns='period',index='etype', values=0)
df2

In [ ]:
#stack plot 그리기
#어떤 컬럼을 stack 할 것인가? >> 행동유형은 종류가 많다. 쿼터 별로 나누자!
df2.plot(kind='___', _______=____, figsize=(18, 6))
plt.title('쿼터별 행동 발생 횟수', size=15)
plt.xticks(rotation=__)

plt.show()

###4-3. 경기 결과 집계하고 시계열 그래프로 그리기

In [ ]:
#날짜별 경기 결과를 group by로 계산해보자
#결과 컬럼에서 성공했을때의 득점만 더하면 경기의 총 점수가 나온다
made = lakers['_____'].____(['made'])
lakers[made].head()

In [ ]:
#날짜별 팀별 슛 성공 건수를 집계해보면?
df3 = lakers[____][['date', 'team', 'points']].________(['____', '____'], as_index=True).___()
df3

In [ ]:
#reset index 사용
df3.___________(inplace=True)
df3

In [ ]:
#LA 레이커스와 상대편으로 경기 결과 나누기
lal = df3[df3['team'] __ 'LAL']
opp = df3[df3['team'] __ 'LAL']

In [ ]:
lal

In [ ]:
#경기 결과를 가로 방향으로 붙이기
#pd.merge 함수 사용: 붙일 때 key는? = 날짜!
df4 = pd._____(___, ___, on='____')
df4

In [ ]:
#La레이커스가 이긴 날과 진 날을 고르자
win = df4[df4['points_x'] _ df4['points_y']]
los = df4[df4['points_x'] _ df4['points_y']]
   
print("08 - 09 시즌에 LA 레이커스는 총 {}회의 경기 중, {}번 우승했다".format(df4._____[0], df4[df4['points_x'] > df4['points_y']]._____[0] ))                         

In [ ]:
#경기 결과를 시계열 그래프로 확인하기
plt.figure(figsize=(16,5))

plt.title("08 - 09 Season : LA Lakers", fontsize=15)

#LA 레이커스 점수를 오렌지 실선으로 나타낸다
plt.plot(lal['____'], lal['_____'], "_", alpha= 1, color = '______', label='LA Lakers')

#LA 레이커스가 이긴 날은 빨간 점, 진 날은 파란 점으로 표시해주자
plt.plot(win['date'], win["points_x"], '_', alpha=.6, color = '___')
plt.plot(los['date'], los["points_x"], '_', alpha=.6, color = '____')

#상대팀 점수를 초록색 점선으로 나타낸다
plt.plot(opp['____'], opp["_______"], "__", alpha=.6, color = '____', label='Opposite team')

plt.legend(loc='upper left', borderaxespad=0, fontsize=13)
plt.xticks(rotation=45)

plt.show()

###4-4. LA레이커스 팀의 코트 위치별 행동

In [ ]:
#LA lakers의 데이터만 가져오기: Team = LAL 인 것
lal2 = lakers[lakers['team'] == '___']
lal2

In [ ]:
#슛한 위치 확인하기
#득점은 어느 위치에서 발생했는가?
sns.set_style('whitegrid')
f, ax = plt.subplots(1, 2, figsize=(12, 7), dpi = 80)

#득점 성공/실패 분포
g1 = sns.________(data=lal2, x='_', y='_', hue='result', _____ = 0.3, palette='Paired', ax=ax[0])
g2 = sns._______(data=lal2, x='_', y='_', hue='result', palette='Paired', ax=ax[0])

#득점 점수 별 분포
g3 = sns.________(data=lal2, x='_', y='_', hue='points', alpha = 0.3, palette='Set3', ax=ax[1])
g4 = sns._______(data=lal2, x='_', y='_', hue='points', palette='Set3', ax=ax[1])

plt.show()
#x, y의 NaN을 0으로 처리해줬기 때문에 0 근처에 쏠림이 생겼다

In [ ]:
#세부 행동 별 슛 위치는 어떻게 될까?
#가장 많이 한 세부 행동 10개 고르기
lal3 = lal2[lal2['type'].____(lal2['type']._________().____(10)._____)]
lal3

In [ ]:
#세부 행동 별 위치 확인하기
sns.set_style('whitegrid')
f, ax = plt.subplots(1, 1, figsize=(5, 7), dpi = 80)

sns._____________(data=lal3, x='_', y='_', hue=lal3['____'], palette='Set3', ax=ax)

#Legend가 10개이기 때문에 그래프 안에 넣으면 혼란스럽다. 옆으로 빼 주자!
ax.______(title='Type', _________________= (1.03, 1))

plt.show()

##5.Review

* LA레이커스 데이터는:  
총 데이터 개수:  450112
총 결측치 수: 76625 = 전체 데이터의 17.02% 
LA레이커스와 경기한 팀 수:  28
경기에 등장하는 행동 수:  10
경기에 등장하는 세부행동 수:  73

* LA레이커스의 홈 경기 vs. 원정경기 비율은?  
  : 홈 경기와 원정경기를 1:1 비율로 치렀다.
* 경기에서 선수들이 가장 많이 하는 행동유형(etype)은?
  : 슛을 가장 많이 하고 그 다음으로는 리바운드.
* 이번 시즌에서 LA레이커스의 경기 결과는?  
  : 08 - 09 시즌에 LA 레이커스는 총 78회의 경기 중, 63번 우승했다.
* LA레이커스 선수들은 코트의 어느 위치에서 어떤 동작을 했는가?
  : 상대편 골대 기준으로 왼쪽에서 슛 했을 때 더 많이 성공했다

* 더 분석해볼 점은?  
  : x, y 좌표의 NaN에 0을 넣지 않는 것이 분석 결과에 더 좋다!  
  : 특정 선수의 행동 패턴 분석
